fhir-resource-tooling/src/main/java/com/ibm/cohort/tooling/fhir/ValueSetImporter.java

```diff
@@ -0,0 +1,161 @@
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+
+public class ValueSetImporter {
+
+	public static final class ValueSetImporterArguments {
+		@Parameter(names = {"-m",
+				"--measure-server"}, description = "Path to JSON configuration data for the FHIR server connection that will be used to retrieve measure and library resources.", required = true)
+		File measureServerConfigFile;
+
+		@Parameter(names = {"-h", "--help"}, description = "Show this help", help = true)
+		boolean isDisplayHelp;
+
+		@Parameter(description = "The list of value set spreadsheets to import", required = true)
+		List<String> spreadsheets;
+	}
+
+	private static class ValueSetArtifact {
+
+		MetadataResource resource;
+		String url;
+		String name;
+		String id;
+	}
+
+	private IGenericClient client;
+	private IParser parser;
+
+	private ValueSetImporter(IGenericClient client) {
+		this.client = client;
+		this.parser = client.getFhirContext().newJsonParser();
+	}
+
+	private void importArtifacts(List<ValueSetArtifact> valueSetArtifacts) {
+
+		for (ValueSetArtifact valueSetArtifact : valueSetArtifacts) {
+			Bundle bundle = client.search().forResource(ValueSet.class).where(ValueSet.URL.matches().value(valueSetArtifact.url))
+					.returnBundle(Bundle.class).execute();
+			if (bundle.getEntry().size() > 0) {
+				valueSetArtifact.id = bundle.getEntryFirstRep().getResource().getIdElement().getIdPart();
+			} else {
+				MethodOutcome outcome = client.create().resource(parser.encodeResourceToString(valueSetArtifact.resource)).execute();
+				if (outcome.getCreated()) {
+					valueSetArtifact.id = outcome.getId().getIdPart();
+				}
+			}
+		}
+	}
+
+	private ValueSetArtifact createArtifact(InputStream is) throws IOException {
+		XSSFWorkbook wb = new XSSFWorkbook(is);
+		XSSFSheet mainSheet = wb.getSheetAt(0);
+		ValueSet valueSet = new ValueSet();
+		for (Row currentRow : mainSheet) {
+			if(currentRow.getCell(0) != null && currentRow.getCell(1) != null) {
+				String value = currentRow.getCell(1).getStringCellValue();
+				switch (currentRow.getCell(0).getStringCellValue()) {
+					case "Value Set Name":
+						valueSet.setName(value);
+						valueSet.setTitle(value);
+						break;
+					case "OID":
+						valueSet.setId(value);
+						String url = "http://cts.nlm.nih.gov/fhir/ValueSet/" + value;
+						valueSet.setUrl(url);
+						break;
+					case "Definition Version":
+						valueSet.setVersion(value);
+						break;
+					default:
+						break;
+				}
+			}
+		}
+		XSSFSheet expansionSheet = wb.getSheetAt(1);
+		ValueSet.ValueSetComposeComponent compose = new ValueSet.ValueSetComposeComponent();
+		List<ValueSet.ConceptSetComponent> composeIncludes = new ArrayList<>();
+		for(Row currentRow : expansionSheet){
+			if (currentRow.getCell(0) != null && currentRow.getCell(0).getCellType().equals(CellType.NUMERIC)){
```

Comment:
I don't believe checking for a cell type of `NUMERIC` is good enough here.
Some codesystems use alphabetical characters in their codes (e.g. ICD-10).

Reply:
I also think we should scan for row with "Code" label in cell 0 as starting point for the list of coded values for this value set - to avoid finding something in the preceding header rows that we interpret as another coded value definition.